### PR TITLE
Add iterator debugging to CI

### PR DESCRIPTION
**Pull request by randombit**
See #3737 #3840

### CODE REVIEW

**Comment by randombit** [comment]
Can't repro the CI failures locally, this may be something in the TPM2 code. Will investigate more later.

File: src/tests/test_bigint.cpp [hunk 1]
```diff
@@ -544,11 +544,17 @@ BOTAN_REGISTER_TEST("math", "bn_rshift", BigInt_Rshift_Test);
...
+   auto rng = Test::new_rng("const_time_left_shift");
+
+   result.start_timer();
 
    Botan::BigInt a = Botan::BigInt::with_capacity(bits / sizeof(Botan::word));
    for(size_t i = 0; i < bits; ++i) {
-      a.set_bit(i);
+      if(rng->next_byte() & 1) {
+         a.set_bit(i);
+      }
```
Comment (reneme) [inline src/tests/test_bigint.cpp:557]: Off topic: should we introduce something like `rng->next_bit() -> CT::Choice`? Not sure whether that would be useful in production code.

**Review by reneme — APPROVED** [review]
Nice! Thanks for picking this up. :)

File: src/tests/test_bigint.cpp (hunk 1 above)
Comment (randombit) [reply src/tests/test_bigint.cpp:557]: Maybe? I can't think of cases currently where we predicate on a single random bit (vs say the even-y bit of a EC point). Somewhere in the PQ algos perhaps?

Comment (reneme) [reply src/tests/test_bigint.cpp:557]: Not really. They all just pull a seed and then sample from KDFs or XOFs. I didn't find something with an ad-hoc search either.

**Pull request merged**